src/molgenis_fdp_harvester/ckan_harvest/baseharvester.py

```diff
@@ -0,0 +1,279 @@
+# SPDX-FileCopyrightText: Open Knowledge
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# This material is copyright (c) Open Knowledge.
+# It is open and licensed under the GNU Affero General Public License (AGPL) v3.0
+# Original location of file: https://github.com/ckan/ckanext-harvest/blob/master/ckanext/harvest/harvesters/base.py
+#
+# Modified by Stichting Health-RI to remove dependencies on CKAN
+
+import logging
+import re
+import uuid
+
+from molgenis_fdp_harvester.ckan_harvest.baseparser import (
+    _munge_to_length,
+    munge_tag,
+    substitute_ascii_equivalents,
+)
+
+
+log = logging.getLogger(__name__)
+
+PACKAGE_NAME_MIN_LENGTH = 2
+PACKAGE_NAME_MAX_LENGTH = 100
+
+
+def munge_title_to_name(name: str) -> str:
+    """Munge a package title into a package name."""
+    name = substitute_ascii_equivalents(name)
+    # convert spaces and separators
+    name = re.sub("[ .:/]", "-", name)
+    # take out not-allowed characters
+    name = re.sub("[^a-zA-Z0-9-_]", "", name).lower()
+    # remove doubles
+    name = re.sub("-+", "-", name)
+    # remove leading or trailing hyphens
+    name = name.strip("-")
+    # if longer than max_length, keep last word if a year
+    max_length = PACKAGE_NAME_MAX_LENGTH - 5
+    # (make length less than max, in case we need a few for '_' chars
+    # to de-clash names.)
+    if len(name) > max_length:
+        year_match = re.match(r".*?[_-]((?:\d{2,4}[-/])?\d{2,4})$", name)
+        if year_match:
+            year = year_match.groups()[0]
+            name = "%s-%s" % (name[: (max_length - len(year) - 1)], year)
+        else:
+            name = name[:max_length]
+    name = _munge_to_length(name, PACKAGE_NAME_MIN_LENGTH, PACKAGE_NAME_MAX_LENGTH)
+    return name
+
+
+class HarvesterBase(object):
+    """
+    Generic base class for harvesters, providing a number of useful functions.
+
+    A harvester doesn't have to derive from this - it could just have:
+
+        implements(IHarvester)
+    """
+
+    config = None
+    _user_name = None
+    _gather_errors = []
+    _harvest_objects = []
+
+    def _save_gather_error(self, *args):
+        log.warning("Harvester gather error: %s", args)
+        self._gather_errors.append(args)
+
+    @classmethod
+    def _gen_new_name(cls, title, existing_name=None, append_type=None):
```

Comment:
If I understand you make first a human friendly dataset  name wich can fit within a url. If that is not unique enough you add an unique number at the end